### PR TITLE
ci(0.76): enable publishing of stable releases

### PR DESCRIPTION
**Pull request by Saadnajmi**
## Summary:

Now that #2308 merged, we have a few followups to actually publish our next stable version:

1. Set `publishTag` to latest in `.ado/templates/npm-publish.yml` (I just missed this)
2. Remove the old publish steps and extra beachball only job, now that we're all in on nx-release. As a consequence, add `react-native-macos-init` to the nx config. 

## Test Plan:

Watch CI after this merges

### CODE REVIEW

**Review by Khalef1 — APPROVED** [review]

File: .ado/publish.yml [hunk 1]
```diff
@@ -74,66 +74,3 @@ extends:
                persistCredentials: true # set to 'true' to leave the OAuth token in the Git config after the initial fetch
 
              - template: /.ado/templates/npm-publish.yml@self
-
-             # Set the git tag and push the version update back to Github
```
Comment (tido64) [inline .ado/publish.yml:78]: We still need to keep this. This pushes changes back to the repo.

Comment (Saadnajmi) [reply .ado/publish.yml:78]: Nx release will do this. I may have accidentally done so (and force push undone it) at least once already 😅

Comment (Saadnajmi) [reply .ado/publish.yml:78]: You can check the (verbose) output of the dry run in my other PR to see when.

Comment (tido64) [reply .ado/publish.yml:78]: Really? The documentation made no mention of it 😛

**Pull request merged**